src/material-experimental/mdc-menu/menu-item.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Component, ChangeDetectionStrategy, ViewEncapsulation} from '@angular/core';
+import {MatMenuItem as BaseMatMenuItem} from '@angular/material/menu';
```

Comment:
Hopefully not, but I'm not quite sure how the inlining is set up. cc @devversion

Reply:
Unfortunately yeah. I'm not sure if there is a way this can be avoided. An idea would be to have a "base" class for the menu item and a class that extends it with the `@Component` decorator. So the the MDC menu item only inherits from the base-class.

Reply:
Wouldn't the `MatMenuItem` still be brought in anyway since we're importing `material/menu`?

Reply:
My thinking is that this would be removed by treeshaking then?

Reply:
I spent some time looking into it, but I don't think that we can avoid not referencing `MatMenuItem` at all, without potentially breaking somebody. We still [need to reference it in here](https://github.com/angular/material2/pull/15857/files#diff-e5383537f481b83ac7838eb77d832bb4R34), in order for DI to work as before and to have the queries on `MatMenu` pick it up. I think that in this case it's fine to take the hit of having the template twice since the entire template is 5 lines of HTML.

Reply:
We could work around that, though, by having `MatMenuItem` being a dummy class that both implementations provide

Reply:
Wouldn't that break consumers that are extending the existing `MatMenuItem`?

Reply:
I don't think so, you could still put all the logic in the base class, and just extend it to tack on the metadata. All the constructor and method signatures would be the same